Tests/Security/Authentication/GoogleApiAuthenticationTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace HP\Bundle\EmailApiBundle\Tests\Security\Authentication;
+
+use HP\Bundle\EmailApiBundle\Security\Authentication\GoogleApiAuthentication;
+use HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\GoogleResourceOwner;
+use HWI\Bundle\OAuthBundle\Security\Core\Authentication\Token\OAuthToken;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Class GoogleApiAuthenticationTest
+ *
+ * Tests the Authentication of the Google API.
+ *
+ * @package HP\Bundle\EmailApiBundle\Tests\Security\Authentication
+ */
+class GoogleApiAuthenticationTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var GoogleApiAuthentication
+     */
+    private $googleApiAuthentication;
+
+    /**
+     * @var GoogleResourceOwner
+     */
+    private $googleResourceOwner;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var string
+     */
+    private $firewallName;
+
+    public function setUp()
```

Comment:
I'm extending \PHPUnit_Framework_TestCase which will execute setUp() each time it runs a test (it is a public method that has a "test" prefix in this name).

If I just wanted one instance of something I could use __construct. Each test in this case is isolated from the others.